### ISSUE TITLE
Provide a way to know who is calling the service

### ISSUE THREAD

**Issue by lucasmoten** [DecipherNow/gm-fabric-go#134]
Within the service operations ... To adequately support both the identity of the user and potential impersonation, it is vital that the fabric support picking up and relaying related headers for user identity, and ideally, any peer certificate information.

Depending on where OAuth goes with tokens, some of this pattern will obviously change in the future, but ultimately, services need to know WHO is making the call, or WHO the call is being made ON BEHALF of, so that they can make authorization decisions and route to dependent services accordingly.

Previously, this pattern was given ... which doesn't work since there while there is a peer on the context, there is no AuthInfo...
```
	p, ok := peer.FromContext(w)
	if ok {
		if p.AuthInfo != nil {
			tlsInfo := p.AuthInfo.(credentials.TLSInfo)
			subject = tlsutil.GetDistinguishedName(tlsInfo.State.VerifiedChains[0][0])
			// ... do stuff with subject (ie, setup clients with impersonation value)
		} else {
			// log or error if dependent on identifying user
		}
	} else {
		// log or error if dependent on identifying user
	}
```

Getting the peer from the context only exposes this potential AuthInfo, as well as the Addr of the peer.

If the USER_DN, EXTERNAL_SYS_DN, SSL_CLIENT_S_DN such headers are expected to be available on the context, then the appropriate helper functions need to be given for pulling those values.

**Comment by alecholmez** [comment]
see #140

**Comment by lucasmoten** [comment]
This issue wasn't fixed by #140. 140 exposes the headers, and provides a sample for retrieving the user_dn header.  However, this issue still stands in terms of needing to know who actually called the service.

To clarify ...

If someone calls a service directly, connecting over TLS, then effectively they dont need to provide headers as the user_dn could be ascertained from the certificate info.
If another backend service makes a rest call through a front end proxy (nginx), then yes, we need access to the same headers, but there is also a need to verify that the connection into this service came from the front end proxy.
Lastly, if another backend service connects directly to the service, then that connection certificate is also needed to verify that they are in the whitelist and allowed to impersonate.

![gm-fabric-go aclrestfilter support - page 1](https://user-images.githubusercontent.com/14304023/36115289-aa9babaa-1000-11e8-8ac1-1fa4b55ef667.png)

**Comment by lucasmoten** [comment]
I'm able to see what is connecting to it once I do the following
1. in settings.toml, set gateway_use_tls and grpc_use_tls to true
2. where getting the peer, the AuthInfo will now have the certificate.

In the individual services that care about authorization, I'm going to begin with an AclRestFilter implementation. Later we can sort out some common way to expose that logic, the same way other stuff is being built outside of fabric for serving docs and multipart file data processing
